gramps/gen/db/bookmarks.py

```diff
@@ -20,35 +20,50 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 #
 
-#-------------------------------------------------------------------------
+from ..utils.callback import Callback
+
+# -------------------------------------------------------------------------
 #
-# class DbBookmarks
+# DbBookmarks class
 #
-#-------------------------------------------------------------------------
-class DbBookmarks:
-    def __init__(self, default=[]):
-        self.bookmarks = list(default) # want a copy (not an alias)
+# -------------------------------------------------------------------------
+class DbBookmarks(Callback):
+
+    __signals__ = {"bookmark-list-changed": (str,)}
+
+    def __init__(self, default=[], cls_type="Person"):
+        super().__init__()
+        self.cls_type = cls_type
+        self.bookmarks = list(default)  # want a copy (not an alias)
 
-    def set(self, new_list):
+    def set(self, new_list, quiet=False):
```

Comment:
Rather than having a quiet mode, the usual way to do this would be to use the `disable_signals()` and `enable_signals()` methods.

Reply:
So if I follow your thinking that would necessitate three lines for each bookmark type when they are loaded by the database class instead of one.  I think this is cleaner., although I suppose a keyword like skip_signal would make intent clearer.

Reply:
Choose to add a load method that doesn't emit and call that from the db class as seems cleaner than this.